cmd/cli_test.go

```diff
@@ -345,6 +333,81 @@ func run(t *testing.T, test testCase) {
 	}
 }
 
+// Simple explanation:
```

Comment:
CLI test will now use a text.template based approach: expected output txt files will have access to template data in the same manner as, for example, our builder's template scripts.

This allow us to execute templates against dynamically loaded data, like list of supported architectures, list of supported targets, current architecture...

@dwindsor i think the same approach could also be used for kernelurls formatting, if that is still a desired feature :)

Reply:
I really like the approach of using `go:embed` in the builders, so yes this is a good change imo!

@FedeDP with respect to `--kernelurls` formatting, what are your thoughts on deprecating all crawling functionality from driverkit altogether? IOW, force folks to use kernel-crawler ;). Only reason I suggest that is, headers package formatting is a hard problem and I feel like it's more maintainable long-term for the project to only have this logic in one place? But, I'm not sure, what are your thoughts? Would this annoy existing users you think?

Reply:
> I really like the approach of using go:embed in the builders, so yes this is a good change imo!

I am using `text.template` here though :D 
> what are your thoughts on deprecating all crawling functionality from driverkit altogether?

Yeah, i think that would be shocking for existing users; i think some kind of basic headers discovery logic should remain nested in driverkit for a while, perhaps we can add a "deprecation" notice in the README?

Reply:
Yeah I was thinking longer term maybe, I'm just thinking about the explosion of distros/builders we're going to (hopefully!) be getting, and the burden of maintaining them in multiple places (new flavors added, vendor repo reorgs, etc). At this point it's not an issue at all, just felt like it was worth calling out.

Reply:
Fully agree with you :/ 
I think that we should properly state that intenral logic is a fallback one, but the preferred method is passing `kernelurls` parameter (be it manually found or taken from kernel-crawler!)

Reply:
I am going to add a NOTE about that!

Reply:
Done!